wrappers/openvino/dnn/rs-dnn-vino.cpp

```diff
@@ -0,0 +1,271 @@
+
+    const auto window_name = "OpenVINO DNN sample";
+    cv::namedWindow( window_name, cv::WINDOW_AUTOSIZE );
+
+    cv::Mat prev_image;
+    openvino_helpers::detected_faces faces;
+    size_t id = 0;
+    uint64 last_frame_number = 0;
+    high_resolution_clock::time_point switch_time = high_resolution_clock::now();
+
+    while( cv::getWindowProperty( window_name, cv::WND_PROP_AUTOSIZE ) >= 0 )
+    {
+        // Wait for the next set of frames
+        auto frames = pipe.wait_for_frames();
+        // Make sure the frames are spatially aligned
+        frames = align_to.process( frames );
+
+        auto color_frame = frames.get_color_frame();
+        auto depth_frame = frames.get_depth_frame();
+
+        // If we only received a new depth frame, but the color did not update, continue
+        if( color_frame.get_frame_number() == last_frame_number )
+            continue;
+        last_frame_number = color_frame.get_frame_number();
+
+        auto image = frame_to_mat( color_frame );
+
+        // We process the previous frame so if this is our first then queue it and continue
+        if( ! p_detector->_request )
+        {
+            p_detector->enqueue( image );
+            p_detector->submit_request();
+            prev_image = image;
+            continue;
+        }
+
+        // Wait for the results of the previous frame we enqueued: we're going to process these
+        p_detector->wait();
+        auto const results = p_detector->fetch_results();
+
+        // Enqueue the current frame so we'd get the results when the next frame comes along!
+        p_detector->enqueue( image );
+        p_detector->submit_request();
+
+        openvino_helpers::detected_faces prev_faces { std::move( faces ) };
+        faces.clear();
+        for( size_t i = 0; i < results.size(); ++i )
+        {
+            auto const & result = results[i];
+            if( result.label <= 0 )
+                continue;  // ignore "background", though not clear why we'd get it
+            cv::Rect rect = result.location;
+            rect = rect & cv::Rect( 0, 0, image.cols, image.rows );
+            auto face_ptr = openvino_helpers::find_face( rect, prev_faces );
+            if( !face_ptr )
+            {
+                // New face
+                std::string label;
+                if( result.label < p_labels->size() )
+                    label = (*p_labels)[result.label];
+                face_ptr = std::make_shared< openvino_helpers::detected_face >( id++, label, rect );
+            }
+            else
+            {
+                // Existing face; just update its parameters
+                face_ptr->move( rect );
+            }
+            faces.push_back( face_ptr );
+        }
+
+        // Keep this image so we can actually process pieces of it once we have the results
+        prev_image = image;
+
+        // Display the results (from the last frame) as rectangles on top (of the current frame)
+        for( auto && face : faces )
```

Comment:
Is it possible to make the main function more concise  so it will focus on the detection flow and wrap 
 utility routines and rendering stuff with function calls?

Reply:
I separated the main body into several functions:

detect_objects()
draw_objects()
draw_detector_overlay()

That does make it better.